src/olympia/lib/crypto/packaged.py

```diff
@@ -172,3 +179,21 @@ def is_signed(file_path):
         filenames = set()
     return set([u'META-INF/mozilla.rsa', u'META-INF/mozilla.sf',
                 u'META-INF/manifest.mf']).issubset(filenames)
+
+
+def get_signature_ou(pkcs7):
+    """
+    Extracts the ou out of a DER formatted, detached PKCS7
+    signature buffer
+    """
+    content = cms.ContentInfo.load(pkcs7)['content'].native
+
+    # Fetch the certificate stack that is the list of signers
+    # Since there should only be one in this use case, take the zeroth
+    # cert in the stack and return its OU
+    try:
+        cert = content['certificates'][0]['tbs_certificate']
```

Comment:
Doesn't seem overly complicated.  @EnTeQuAk do you want to move the function into signing_clients as is or should we future proof it now?